src/main/java/co/inventorsoft/academy/collections/model/Range.java

```diff
@@ -3,58 +3,99 @@
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Function;
 
 public class Range<T> implements Set<T> {
 
+    private TreeSet<T> data;
+
+    public Range() {
+        data = new TreeSet<>();
+    }
+
+    public static Range<Integer> of(int e1, int e2){
+        Range<Integer> range = new Range<>();
+        if (e1 < e2){
+            for (Integer i = e1; i <= e2; i++){
+                range.add(i);
+            }
+        }
+        return range;
+    }
+
+    public static Range<Float> of(float e1, float e2){
+        Range<Float> range = new Range<>();
+        if (e1 < e2){
+            for (Float i = e1; i <= e2; i = i + 0.1f){
+                range.add(i);
+            }
+        }
+        return range;
+    }
+
+    public static Range<Character> of(char e1, char e2, Function function){
```

Comment:
1. Add support for custom type.
2. After, remove redundant code in methods **of** there is similar code.

Reply:
Not sure if you are notified, but i fixed stuff you pointed me at (quite a while ago, actually).
Should i "resolve" this conversation, or is it up to you?